### PR TITLE
Added Hitec-Airspeed AP_Periph target

### PR DESCRIPTION
**Pull request by tridge**
This uses a new option to list specific airspeed sensors in hwdef.dat to save flash space

### CODE REVIEW

File: libraries/AP_Airspeed/AP_Airspeed.cpp [hunk 1]
```diff
@@ -294,6 +294,36 @@ AP_Airspeed::AP_Airspeed()
...
+// macro for use by HAL_INS_PROBE_LIST
+#define GET_I2C_DEVICE(bus, address) hal.i2c_mgr->get_device(bus, address)
+
+bool AP_Airspeed::add_backend(AP_Airspeed_Backend *backend)
+{
+    if (!backend) {
+        return false;
+    }
+    if (num_sensors >= AIRSPEED_MAX_SENSORS) {
+        AP_HAL::panic("Too many airspeed drivers");
```
Comment (WickedShell) [inline libraries/AP_Airspeed/AP_Airspeed.cpp:306]: This is great and all, but using the `ADD_BACKEND` macro means you will almost never see this message, and instead we will just not create the sensor, and won't emit any form of warning.

**Review by WickedShell — CHANGES_REQUESTED** [review]
How is this superior then setting the airspeed parameter with the default param file? It seems like I'm missing something, otherwise this surely seems like more flash then one more default param entry...

**Comment by tridge** [comment]
> It seems like I'm missing something, otherwise this surely seems like more flash then one more default param entry...

it saves linking all of the drivers as the switch statement is removed

File: libraries/AP_HAL_ChibiOS/hwdef/scripts/chibios_hwdef.py [hunk 2]
```diff
@@ -1260,6 +1261,37 @@ def write_BARO_config(f):
...
+                dev[i] = parse_spi_device(dev[i])
+            elif dev[i].startswith("I2C:"):
+                (wrapper, dev[i]) = parse_i2c_device(dev[i])
+                if dev[i].startswith('hal.i2c_mgr'):
+                    dev[i] = 'std::move(%s)' % dev[i]
+        n = len(devlist)+1
+        devlist.append('HAL_AIRSPEED_PROBE%u' % n)
+        args = ['*this', str(idx)] + dev[1:]
+        f.write(
+            '#define HAL_AIRSPEED_PROBE%u %s ADD_BACKEND(AP_Airspeed_%s::probe(%s))\n'
```
[the remark this answers is not in the document]
Comment (tridge) [reply libraries/AP_HAL_ChibiOS/hwdef/scripts/chibios_hwdef.py:1289]: it is the same pattern we use for IMUs, baros, compasses

File: libraries/AP_Airspeed/AP_Airspeed.cpp (hunk 1 above)
Comment (tridge) [reply libraries/AP_Airspeed/AP_Airspeed.cpp:306]: which is the same as compass, baro and imu
a message on a peripheral isn't all that useful

File: libraries/AP_Airspeed/AP_Airspeed_DLVR.cpp [hunk 3]
```diff
@@ -36,13 +36,30 @@ AP_Airspeed_DLVR::AP_Airspeed_DLVR(AP_Airspeed &_frontend, uint8_t _instance, co
...
 {
-    dev = hal.i2c_mgr->get_device(get_bus(), DLVR_I2C_ADDR);
-    if (!dev) {
-        return false;
+    if (!_dev) {
+        return nullptr;
+    }
+    AP_Airspeed_DLVR *sensor = new AP_Airspeed_DLVR(_frontend, _instance, _range_inH2O);
+    if (!sensor) {
+        delete sensor;
```
[the remark this answers is not in the document]
Comment (tridge) [reply libraries/AP_Airspeed/AP_Airspeed_DLVR.cpp:52]: yes, its harmless, but can be removed

[the remark this answers is not in the document]
Comment (tridge) [reply libraries/AP_Airspeed/AP_Airspeed_DLVR.cpp]: fixed

**Comment by tridge** [comment]
@WickedShell the ADD_BACKEND() pattern with hidden probe() method is a bit of a trap, but is also our established pattern. It is better than the huge ifdef tree for each board we used to have

File: libraries/AP_Airspeed/AP_Airspeed.cpp [hunk 4]
```diff
@@ -294,6 +294,36 @@ AP_Airspeed::AP_Airspeed()
     _singleton = this;
 }
 
+// macro for use by HAL_INS_PROBE_LIST
```
Comment (peterbarker) [inline libraries/AP_Airspeed/AP_Airspeed.cpp:297]:
```suggestion
// macro for use by HAL_AIRSPEED_PROBE_LIST
```

**Review by peterbarker — APPROVED** [review]
As mentioned on the DevCall, at least we'll be consistently bad.  Meaning at some stage we might be able to change things to be consistently good.

**Pull request merged**